// ews-javascript-api/js/MailboxSearch/SearchMailboxesResult.js
"use strict";
var SearchMailboxesResult = (function () {
    function SearchMailboxesResult() {
    }
    SearchMailboxesResult.prototype.GetRecipients = function (reader, elementName) { throw new Error("SearchMailboxesResult.ts - GetRecipients : Not implemented."); };
    SearchMailboxesResult.prototype.LoadExtendedPropertiesXml = function (reader) { throw new Error("SearchMailboxesResult.ts - LoadExtendedPropertiesXml : Not implemented."); };
    SearchMailboxesResult.prototype.LoadFromJson = function (jsonObject) { throw new Error("SearchMailboxesResult.ts - LoadFromJson : Not implemented."); };
    SearchMailboxesResult.prototype.LoadFromXml = function (reader) { throw new Error("SearchMailboxesResult.ts - LoadFromXml : Not implemented."); };
    SearchMailboxesResult.prototype.LoadKeywordStatsXml = function (reader) { throw new Error("SearchMailboxesResult.ts - LoadKeywordStatsXml : Not implemented."); };
    SearchMailboxesResult.prototype.LoadPreviewItemsXml = function (reader) { throw new Error("SearchMailboxesResult.ts - LoadPreviewItemsXml : Not implemented."); };
    return SearchMailboxesResult;
}());
exports.SearchMailboxesResult = SearchMailboxesResult;
//}
