// ews-javascript-api/js/Core/Requests/ApplyConversationActionRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var MultiResponseServiceRequest_1 = require("./MultiResponseServiceRequest");
/**
 * ## @internal *Not Implemented*
 */
var ApplyConversationActionRequest = (function (_super) {
    __extends(ApplyConversationActionRequest, _super);
    function ApplyConversationActionRequest() {
        _super.apply(this, arguments);
    }
    ApplyConversationActionRequest.prototype.CreateServiceResponse = function (service, responseIndex) { throw new Error("ApplyConversationActionRequest.ts - CreateServiceResponse : Not implemented."); };
    ApplyConversationActionRequest.prototype.GetExpectedResponseMessageCount = function () { throw new Error("ApplyConversationActionRequest.ts - GetExpectedResponseMessageCount : Not implemented."); };
    ApplyConversationActionRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("ApplyConversationActionRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    ApplyConversationActionRequest.prototype.GetResponseMessageXmlElementName = function () { throw new Error("ApplyConversationActionRequest.ts - GetResponseMessageXmlElementName : Not implemented."); };
    ApplyConversationActionRequest.prototype.GetResponseXmlElementName = function () { throw new Error("ApplyConversationActionRequest.ts - GetResponseXmlElementName : Not implemented."); };
    ApplyConversationActionRequest.prototype.GetXmlElementName = function () { throw new Error("ApplyConversationActionRequest.ts - GetXmlElementName : Not implemented."); };
    ApplyConversationActionRequest.prototype.Validate = function () { throw new Error("ApplyConversationActionRequest.ts - Validate : Not implemented."); };
    ApplyConversationActionRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("ApplyConversationActionRequest.ts - WriteElementsToXml : Not implemented."); };
    return ApplyConversationActionRequest;
}(MultiResponseServiceRequest_1.MultiResponseServiceRequest));
exports.ApplyConversationActionRequest = ApplyConversationActionRequest;
