// ews-javascript-api/js/Core/Requests/DisableAppRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/**
 * ## @internal *Not Implemented*
 */
var DisableAppRequest = (function (_super) {
    __extends(DisableAppRequest, _super);
    function DisableAppRequest() {
        _super.apply(this, arguments);
    }
    DisableAppRequest.prototype.Execute = function () { throw new Error("DisableAppRequest.ts - Execute : Not implemented."); };
    DisableAppRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("DisableAppRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    DisableAppRequest.prototype.GetResponseXmlElementName = function () { throw new Error("DisableAppRequest.ts - GetResponseXmlElementName : Not implemented."); };
    DisableAppRequest.prototype.GetXmlElementName = function () { throw new Error("DisableAppRequest.ts - GetXmlElementName : Not implemented."); };
    DisableAppRequest.prototype.ParseResponse = function (reader) { throw new Error("DisableAppRequest.ts - ParseResponse : Not implemented."); };
    DisableAppRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("DisableAppRequest.ts - WriteElementsToXml : Not implemented."); };
    return DisableAppRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.DisableAppRequest = DisableAppRequest;
