// ews-javascript-api/js/Core/Requests/FindConversationRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/**
 * ## @internal *Not Implemented*
 */
var FindConversationRequest = (function (_super) {
    __extends(FindConversationRequest, _super);
    function FindConversationRequest() {
        _super.apply(this, arguments);
    }
    FindConversationRequest.prototype.Execute = function () { throw new Error("FindConversationRequest.ts - Execute : Not implemented."); };
    FindConversationRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("FindConversationRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    FindConversationRequest.prototype.GetResponseXmlElementName = function () { throw new Error("FindConversationRequest.ts - GetResponseXmlElementName : Not implemented."); };
    FindConversationRequest.prototype.GetXmlElementName = function () { throw new Error("FindConversationRequest.ts - GetXmlElementName : Not implemented."); };
    FindConversationRequest.prototype.ParseResponse = function (reader) { throw new Error("FindConversationRequest.ts - ParseResponse : Not implemented."); };
    //ParseResponse(jsonBody: JsonObject): any { throw new Error("FindConversationRequest.ts - ParseResponse : Not implemented."); }
    FindConversationRequest.prototype.Validate = function () { throw new Error("FindConversationRequest.ts - Validate : Not implemented."); };
    FindConversationRequest.prototype.WriteAttributesToXml = function (writer) { throw new Error("FindConversationRequest.ts - WriteAttributesToXml : Not implemented."); };
    FindConversationRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("FindConversationRequest.ts - WriteElementsToXml : Not implemented."); };
    return FindConversationRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.FindConversationRequest = FindConversationRequest;
