// ews-javascript-api/js/Core/Requests/GetAppMarketplaceUrlRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/**
 * ## @internal *Not Implemented*
 */
var GetAppMarketplaceUrlRequest = (function (_super) {
    __extends(GetAppMarketplaceUrlRequest, _super);
    function GetAppMarketplaceUrlRequest() {
        _super.apply(this, arguments);
    }
    GetAppMarketplaceUrlRequest.prototype.Execute = function () { throw new Error("GetAppMarketplaceUrlRequest.ts - Execute : Not implemented."); };
    GetAppMarketplaceUrlRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("GetAppMarketplaceUrlRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    GetAppMarketplaceUrlRequest.prototype.GetResponseXmlElementName = function () { throw new Error("GetAppMarketplaceUrlRequest.ts - GetResponseXmlElementName : Not implemented."); };
    GetAppMarketplaceUrlRequest.prototype.GetXmlElementName = function () { throw new Error("GetAppMarketplaceUrlRequest.ts - GetXmlElementName : Not implemented."); };
    GetAppMarketplaceUrlRequest.prototype.ParseResponse = function (reader) { throw new Error("GetAppMarketplaceUrlRequest.ts - ParseResponse : Not implemented."); };
    GetAppMarketplaceUrlRequest.prototype.Validate = function () { throw new Error("GetAppMarketplaceUrlRequest.ts - Validate : Not implemented."); };
    GetAppMarketplaceUrlRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("GetAppMarketplaceUrlRequest.ts - WriteElementsToXml : Not implemented."); };
    return GetAppMarketplaceUrlRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.GetAppMarketplaceUrlRequest = GetAppMarketplaceUrlRequest;
