// ews-javascript-api/js/Core/Requests/GetAppManifestsRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/** @internal */
var GetAppManifestsRequest = (function (_super) {
    __extends(GetAppManifestsRequest, _super);
    function GetAppManifestsRequest() {
        _super.apply(this, arguments);
    }
    GetAppManifestsRequest.prototype.Execute = function () { throw new Error("GetAppManifestsRequest.ts - Execute : Not implemented."); };
    GetAppManifestsRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("GetAppManifestsRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    GetAppManifestsRequest.prototype.GetResponseXmlElementName = function () { throw new Error("GetAppManifestsRequest.ts - GetResponseXmlElementName : Not implemented."); };
    GetAppManifestsRequest.prototype.GetXmlElementName = function () { throw new Error("GetAppManifestsRequest.ts - GetXmlElementName : Not implemented."); };
    GetAppManifestsRequest.prototype.ParseResponse = function (reader) { throw new Error("GetAppManifestsRequest.ts - ParseResponse : Not implemented."); };
    GetAppManifestsRequest.prototype.Validate = function () { throw new Error("GetAppManifestsRequest.ts - Validate : Not implemented."); };
    GetAppManifestsRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("GetAppManifestsRequest.ts - WriteElementsToXml : Not implemented."); };
    return GetAppManifestsRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.GetAppManifestsRequest = GetAppManifestsRequest;
