// ews-javascript-api/js/Core/Requests/GetConversationItemsRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var MultiResponseServiceRequest_1 = require("./MultiResponseServiceRequest");
/**
 * ## @internal *Not Implemented*
 */
var GetConversationItemsRequest = (function (_super) {
    __extends(GetConversationItemsRequest, _super);
    function GetConversationItemsRequest() {
        _super.apply(this, arguments);
    }
    GetConversationItemsRequest.prototype.CreateServiceResponse = function (service, responseIndex) { throw new Error("GetConversationItemsRequest.ts - CreateServiceResponse : Not implemented."); };
    GetConversationItemsRequest.prototype.GetExpectedResponseMessageCount = function () { throw new Error("GetConversationItemsRequest.ts - GetExpectedResponseMessageCount : Not implemented."); };
    GetConversationItemsRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("GetConversationItemsRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    GetConversationItemsRequest.prototype.GetResponseMessageXmlElementName = function () { throw new Error("GetConversationItemsRequest.ts - GetResponseMessageXmlElementName : Not implemented."); };
    GetConversationItemsRequest.prototype.GetResponseXmlElementName = function () { throw new Error("GetConversationItemsRequest.ts - GetResponseXmlElementName : Not implemented."); };
    GetConversationItemsRequest.prototype.GetXmlElementName = function () { throw new Error("GetConversationItemsRequest.ts - GetXmlElementName : Not implemented."); };
    GetConversationItemsRequest.prototype.Validate = function () { throw new Error("GetConversationItemsRequest.ts - Validate : Not implemented."); };
    GetConversationItemsRequest.prototype.WriteAttributesToXml = function (writer) { throw new Error("GetConversationItemsRequest.ts - WriteAttributesToXml : Not implemented."); };
    GetConversationItemsRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("GetConversationItemsRequest.ts - WriteElementsToXml : Not implemented."); };
    return GetConversationItemsRequest;
}(MultiResponseServiceRequest_1.MultiResponseServiceRequest));
exports.GetConversationItemsRequest = GetConversationItemsRequest;
