// ews-javascript-api/js/Core/Requests/GetUserRetentionPolicyTagsRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/**
 * ## @internal *Not Implemented*
 */
var GetUserRetentionPolicyTagsRequest = (function (_super) {
    __extends(GetUserRetentionPolicyTagsRequest, _super);
    function GetUserRetentionPolicyTagsRequest() {
        _super.apply(this, arguments);
    }
    GetUserRetentionPolicyTagsRequest.prototype.Execute = function () { throw new Error("GetUserRetentionPolicyTagsRequest.ts - Execute : Not implemented."); };
    GetUserRetentionPolicyTagsRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("GetUserRetentionPolicyTagsRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    GetUserRetentionPolicyTagsRequest.prototype.GetResponseXmlElementName = function () { throw new Error("GetUserRetentionPolicyTagsRequest.ts - GetResponseXmlElementName : Not implemented."); };
    GetUserRetentionPolicyTagsRequest.prototype.GetXmlElementName = function () { throw new Error("GetUserRetentionPolicyTagsRequest.ts - GetXmlElementName : Not implemented."); };
    GetUserRetentionPolicyTagsRequest.prototype.ParseResponse = function (reader) { throw new Error("GetUserRetentionPolicyTagsRequest.ts - ParseResponse : Not implemented."); };
    GetUserRetentionPolicyTagsRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("GetUserRetentionPolicyTagsRequest.ts - WriteElementsToXml : Not implemented."); };
    return GetUserRetentionPolicyTagsRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.GetUserRetentionPolicyTagsRequest = GetUserRetentionPolicyTagsRequest;
