// ews-javascript-api/js/Core/Requests/InstallAppRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/**
 * ## @internal *Not Implemented*
 */
var InstallAppRequest = (function (_super) {
    __extends(InstallAppRequest, _super);
    function InstallAppRequest() {
        _super.apply(this, arguments);
    }
    InstallAppRequest.prototype.Execute = function () { throw new Error("InstallAppRequest.ts - Execute : Not implemented."); };
    InstallAppRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("InstallAppRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    InstallAppRequest.prototype.GetResponseXmlElementName = function () { throw new Error("InstallAppRequest.ts - GetResponseXmlElementName : Not implemented."); };
    InstallAppRequest.prototype.GetXmlElementName = function () { throw new Error("InstallAppRequest.ts - GetXmlElementName : Not implemented."); };
    InstallAppRequest.prototype.ParseResponse = function (reader) { throw new Error("InstallAppRequest.ts - ParseResponse : Not implemented."); };
    InstallAppRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("InstallAppRequest.ts - WriteElementsToXml : Not implemented."); };
    return InstallAppRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.InstallAppRequest = InstallAppRequest;
