// ews-javascript-api/js/Core/Requests/UninstallAppRequest.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var SimpleServiceRequestBase_1 = require("./SimpleServiceRequestBase");
/**
 * ## @internal *Not Implemented*
 */
var UninstallAppRequest = (function (_super) {
    __extends(UninstallAppRequest, _super);
    function UninstallAppRequest() {
        _super.apply(this, arguments);
    }
    UninstallAppRequest.prototype.Execute = function () { throw new Error("UninstallAppRequest.ts - Execute : Not implemented."); };
    UninstallAppRequest.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("UninstallAppRequest.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    UninstallAppRequest.prototype.GetResponseXmlElementName = function () { throw new Error("UninstallAppRequest.ts - GetResponseXmlElementName : Not implemented."); };
    UninstallAppRequest.prototype.GetXmlElementName = function () { throw new Error("UninstallAppRequest.ts - GetXmlElementName : Not implemented."); };
    UninstallAppRequest.prototype.ParseResponse = function (reader) { throw new Error("UninstallAppRequest.ts - ParseResponse : Not implemented."); };
    UninstallAppRequest.prototype.WriteElementsToXml = function (writer) { throw new Error("UninstallAppRequest.ts - WriteElementsToXml : Not implemented."); };
    return UninstallAppRequest;
}(SimpleServiceRequestBase_1.SimpleServiceRequestBase));
exports.UninstallAppRequest = UninstallAppRequest;
