// ews-javascript-api/js/Core/ServiceObjects/ResponseObjects/PostReply.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var XmlElementNames_1 = require("../../XmlElementNames");
var ServiceObject_1 = require("../ServiceObject");
var PostReply = (function (_super) {
    __extends(PostReply, _super);
    function PostReply() {
        _super.apply(this, arguments);
    }
    PostReply.prototype.GetMinimumRequiredServerVersion = function () { throw new Error("PostReply.ts - GetMinimumRequiredServerVersion : Not implemented."); };
    PostReply.prototype.GetSchema = function () { throw new Error("PostReply.ts - GetSchema : Not implemented."); };
    PostReply.prototype.GetXmlElementName = function () { return XmlElementNames_1.XmlElementNames.PostReplyItem; };
    PostReply.prototype.InternalCreate = function (parentFolderId, messageDisposition) { throw new Error("PostReply.ts - InternalCreate : Not implemented."); };
    PostReply.prototype.InternalDelete = function (deleteMode, sendCancellationsMode, affectedTaskOccurrences) { throw new Error("PostReply.ts - InternalDelete : Not implemented."); };
    PostReply.prototype.InternalLoad = function (propertySet) { throw new Error("PostReply.ts - InternalLoad : Not implemented."); };
    //Save(): PostItem { throw new Error("PostReply.ts - Save : Not implemented."); }
    //Save(destinationFolderId: FolderId): PostItem { throw new Error("PostReply.ts - Save : Not implemented."); }
    PostReply.prototype.Save = function (destinationFolderName) { throw new Error("PostReply.ts - Save : Not implemented."); };
    return PostReply;
}(ServiceObject_1.ServiceObject));
exports.PostReply = PostReply;
//
//}
