// ews-javascript-api/js/Search/ConversationIndexedItemView.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var PagedView_1 = require("./PagedView");
var ConversationIndexedItemView = (function (_super) {
    __extends(ConversationIndexedItemView, _super);
    function ConversationIndexedItemView() {
        _super.apply(this, arguments);
    }
    ConversationIndexedItemView.prototype.AddJsonProperties = function (jsonRequest, service) { throw new Error("ConversationIndexedItemView.ts - AddJsonProperties : Not implemented."); };
    ConversationIndexedItemView.prototype.GetServiceObjectType = function () { throw new Error("ConversationIndexedItemView.ts - GetServiceObjectType : Not implemented."); };
    ConversationIndexedItemView.prototype.GetViewJsonTypeName = function () { throw new Error("ConversationIndexedItemView.ts - GetViewJsonTypeName : Not implemented."); };
    ConversationIndexedItemView.prototype.GetViewXmlElementName = function () { throw new Error("ConversationIndexedItemView.ts - GetViewXmlElementName : Not implemented."); };
    /** @internal */
    ConversationIndexedItemView.prototype.InternalValidate = function (request) { throw new Error("ConversationIndexedItemView.ts - InternalValidate : Not implemented."); };
    ConversationIndexedItemView.prototype.InternalWriteSearchSettingsToXml = function (writer, groupBy) { throw new Error("ConversationIndexedItemView.ts - InternalWriteSearchSettingsToXml : Not implemented."); };
    ConversationIndexedItemView.prototype.WriteAttributesToXml = function (writer) { throw new Error("ConversationIndexedItemView.ts - WriteAttributesToXml : Not implemented."); };
    ConversationIndexedItemView.prototype.WriteOrderByToXml = function (writer) { throw new Error("ConversationIndexedItemView.ts - WriteOrderByToXml : Not implemented."); };
    ConversationIndexedItemView.prototype.WriteToXml = function (writer, groupBy) { throw new Error("ConversationIndexedItemView.ts - WriteToXml : Not implemented."); };
    return ConversationIndexedItemView;
}(PagedView_1.PagedView));
exports.ConversationIndexedItemView = ConversationIndexedItemView;
