// ews-javascript-api/js/Search/ItemView.js
"use strict";
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var EnumToExchangeVersionMappingHelper_1 = require("../Enumerations/EnumToExchangeVersionMappingHelper");
var OffsetBasePoint_1 = require("../Enumerations/OffsetBasePoint");
var XmlElementNames_1 = require("../Core/XmlElementNames");
var EwsUtilities_1 = require("../Core/EwsUtilities");
var XmlAttributeNames_1 = require("../Core/XmlAttributeNames");
var PagedView_1 = require("./PagedView");
var ItemTraversal_1 = require("../Enumerations/ItemTraversal");
var OrderByCollection_1 = require("./OrderByCollection");
var ServiceObjectType_1 = require("../Enumerations/ServiceObjectType");
var ItemView = (function (_super) {
    __extends(ItemView, _super);
    function ItemView(pageSize, offset, offsetBasePoint) {
        if (offset === void 0) { offset = 0; }
        if (offsetBasePoint === void 0) { offsetBasePoint = OffsetBasePoint_1.OffsetBasePoint.Beginning; }
        _super.call(this, pageSize, offset, offsetBasePoint);
        this.Traversal = ItemTraversal_1.ItemTraversal.Shallow;
        //private traversal: ItemTraversal; - not needed 
        this.orderBy = new OrderByCollection_1.OrderByCollection();
    }
    Object.defineProperty(ItemView.prototype, "OrderBy", {
        get: function () { return this.orderBy; },
        enumerable: true,
        configurable: true
    });
    ItemView.prototype.AddJsonProperties = function (jsonRequest /*JsonObject*/, service) { throw new Error("ItemView.ts - AddJsonProperties : Not implemented."); };
    ItemView.prototype.GetServiceObjectType = function () { return ServiceObjectType_1.ServiceObjectType.Item; };
    ItemView.prototype.GetViewJsonTypeName = function () { throw new Error("ItemView.ts - GetViewJsonTypeName : Not implemented."); };
    ItemView.prototype.GetViewXmlElementName = function () { return XmlElementNames_1.XmlElementNames.IndexedPageItemView; };
    /**@internal */
    ItemView.prototype.InternalValidate = function (request) {
        _super.prototype.InternalValidate.call(this, request);
        EwsUtilities_1.EwsUtilities.ValidateEnumVersionValue(EnumToExchangeVersionMappingHelper_1.EnumToExchangeVersionMappingHelper.ItemTraversal, this.Traversal, request.Service.RequestedServerVersion);
    };
    ItemView.prototype.InternalWriteSearchSettingsToXml = function (writer, groupBy) { _super.prototype.InternalWriteSearchSettingsToXml.call(this, writer, groupBy); };
    ItemView.prototype.WriteAttributesToXml = function (writer) {
        writer.WriteAttributeValue(XmlAttributeNames_1.XmlAttributeNames.Traversal, ItemTraversal_1.ItemTraversal[this.Traversal]);
    };
    ItemView.prototype.WriteOrderByToXml = function (writer) { this.orderBy.WriteToXml(writer, XmlElementNames_1.XmlElementNames.SortOrder); };
    return ItemView;
}(PagedView_1.PagedView));
exports.ItemView = ItemView;
